code/headers/mp3_player.hpp

```diff
@@ -0,0 +1,170 @@
+        void init();
+
+        /**
+         * This function simulates a button press for a given amount of
+         * milliseconds.
+         *
+         * @internal
+         * @param milliseconds The amount of milliseconds you want the pin to be
+         * pressed.
+         * @param pin The pin used for the press simulation.
+         */
+        void press_for(unsigned int milliseconds, hwlib::pin_oc &pin);
+
+        /**
+         * This function simulates a 2 button press simultaniusly for a given
+         * amount of milliseconds.
+         *
+         * @internal
+         * @param milliseconds The amount of milliseconds you want the pins to
+         * be pressed.
+         * @param 2 pins used for the press simulation.
+         */
+        void press_for(unsigned int milliseconds, hwlib::pin_oc &pin1,
+                       hwlib::pin_oc &pin2);
+
+        /**
+         * This function uses the press_for function to output a short press
+         * command.
+         *
+         * @internal
+         * @param pin The pin you want to use for the short press.
+         */
+        void short_press(hwlib::pin_oc &pin);
+
+        /**
+         * This function uses the press_for function to output a long press
+         * command for 2 pins at the same time.
+         *
+         * @internal
+         * @param 2 pins you want to use for the long press.
+         */
+        void short_press(hwlib::pin_oc &pin1, hwlib::pin_oc &pin2);
+
+        /**
+         * This function uses the press_for function to output a long press
+         * command.
+         *
+         * @internal
+         * @param pin The pin you want to use for the long press.
+         */
+        void long_press(hwlib::pin_oc &pin);
+
+        /**
+         * This function uses the press_for function to output a long press
+         * command for 2 pins at the same time.
+         *
+         * @internal
+         * @param 2 pins you want to use for the long press.
+         */
+        void long_press(hwlib::pin_oc &pin1, hwlib::pin_oc &pin2);
+
+    public:
+        /**
+         * This function constructs the mp3_player class.
+         *
+         * @param volume_up_next_pin The pin you want to use to control the
+         * volume up and next.
+         * @param volume_down_prev_pin The pin you want to use to control the
+         * volume down and previous.
+         * @param play_pause_mode_pin The pin you want to use to control the
+         * play, pause and mode.
+         * @param repeat_pin The pin you want to use to set on repeat.
+         * @param is_playing_pin The pin you want to use to check if there is a
+         * song playing.
+         */
+        mp3_player_c(hwlib::pin_out &power_pin,
+                     hwlib::pin_oc &volume_up_next_pin,
+                     hwlib::pin_oc &volume_down_prev_pin,
+                     hwlib::pin_oc &play_pause_mode_pin,
+                     hwlib::pin_oc &repeat_pin);
+        // virtual ~mp3_player_c();
```

Comment:
Dit moet of verwijderd worden of uitgecomment lijkt mij

Reply:
Wat bedoel je?

Reply:
dit is een destructor die nu in comments staat het lijkt me dat ie of nuttig is en dus gewoon als code gezien wordt of dat ie weg kan

Reply:
je hebt een virtual destructor op een bepaald moment geschreven, niet verwijderd maar het wordt ook niet gebruikt. 
Alles dat naar de master gaat hoort functioneel te zijn, dus een stuk becommentarieerde code is iets dat daar niet thuis hoort. Of het moet geïmplementeerd zijn, of het moet verwijderd worden.

Reply:
Oooh de deconstructor. Ja die kan weg was me niet opgevallen.